README.md

```diff
@@ -251,3 +251,24 @@ it needs to run as a plugin to Anki desktop to generate notes to a local
 collection. It does not seem to allow editing/adding to a local collection
 outside of Anki desktop.
 
+## Contributing
+
+### Setup
+
+First fork the repository, then clone your fork and install the package
+dependencies:
+
+```console
+$ git clone git@github.com:<username>/apy.git
+$ cd apy/
+$ mkvirtualenv anki -p python3.8
+$ echo "anki" > .venv  # if you use a loader like zsh-autoswitch-virtualenv
+$ ./setup.py install
+$ pip install -r requirements-dev.txt
+```
+
+### tests
+
+```console
+$ pytest
```

Comment:
I'm afraid I couldn't run the tests succesfully, probably not using the right version for some dependency:

```console
[I] ➜ pytest
===================================== test session starts =====================================
platform linux -- Python 3.8.5, pytest-6.1.0, py-1.9.0, pluggy-0.13.1
rootdir: /home/dori/Projects/3rdpart/apy, configfile: pytest.ini
collected 10 items

tests/test_basics.py ...                                                                [ 30%]
tests/test_batch_edit.py .                                                              [ 40%]
tests/test_cli.py Fatal Python error: Aborted

Current thread 0x00007f70e4ee9740 (most recent call first):
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/aqt/profiles.py", line 173 in load
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/apy-0.7.2-py3.8.egg/apy/anki.py", line 63 in _init_load_collection
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/apy-0.7.2-py3.8.egg/apy/anki.py", line 26 in __init__
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/apy-0.7.2-py3.8.egg/apy/cli.py", line 177 in info
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/click/core.py", line 610 in invoke
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/apy-0.7.2-py3.8.egg/apy/cli.py", line 54 in main
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/click/decorators.py", line 21 in new_func
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/click/core.py", line 610 in invoke
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/click/core.py", line 1066 in invoke
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/click/core.py", line 1236 in invoke
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/click/core.py", line 782 in main
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/click/testing.py", line 329 in invoke
  File "/home/dori/Projects/3rdpart/apy/tests/test_cli.py", line 25 in test_cli_base_directory
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/_pytest/python.py", line 184 in pytest_pyfunc_call
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/_pytest/python.py", line 1627 in runtest
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/_pytest/runner.py", line 163 in pytest_runtest_call
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/_pytest/runner.py", line 256 in <lambda>
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/_pytest/runner.py", line 310 in from_call
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/_pytest/runner.py", line 255 in call_runtest_hook
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/_pytest/runner.py", line 216 in call_and_report
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/_pytest/runner.py", line 127 in runtestprotocol
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/_pytest/runner.py", line 110 in pytest_runtest_protocol
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/_pytest/main.py", line 338 in pytest_runtestloop
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/_pytest/main.py", line 313 in _main
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/_pytest/main.py", line 257 in wrap_session
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/_pytest/main.py", line 306 in pytest_cmdline_main
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/_pytest/config/__init__.py", line 164 in main
  File "/home/dori/.virtualenvs/anki/lib/python3.8/site-packages/_pytest/config/__init__.py", line 187 in console_main
  File "/home/dori/.virtualenvs/anki/bin/pytest", line 8 in <module>
zsh: abort (core dumped)  pytest
```

If adding `@pytest.mark.skip` to the `test_cli.py` tests, I get:
```console
[I] ➜ pytest
===================================== test session starts =====================================
platform linux -- Python 3.8.5, pytest-6.1.0, py-1.9.0, pluggy-0.13.1
rootdir: /home/dori/Projects/3rdpart/apy, configfile: pytest.ini
collected 10 items

tests/test_basics.py ...                                                                [ 30%]
tests/test_batch_edit.py .                                                              [ 40%]
tests/test_cli.py sss                                                                   [ 70%]
tests/test_decks.py .                                                                   [ 80%]
tests/test_errors.py .                                                                  [ 90%]
tests/test_models.py .                                                                  [100%]

================================ 7 passed, 3 skipped in 0.35s =================================
```

Reply:
The CLI tests were introduced by someone else, I accepted it because it looked convenient. I actually find testing of apy quite difficult, and I'm sorry to say the test suite is not really good enough. At the moment, I tend to prioritize code improvements and features before proper testing (except manual testing, that is).